examples/x86_64.swiftinterface

```diff
@@ -0,0 +1,80 @@
+}
+@objc @objcMembers public class MicrosoftTeamsUserIdentifier : ObjectiveC.NSObject, AzureCommunication.CommunicationIdentifier {
+  @objc final public let userId: Swift.String
+  @objc final public let isAnonymous: Swift.Bool
+  @objc final public let rawId: Swift.String?
+  @objc final public let cloudEnviroment: AzureCommunication.CommunicationCloudEnvironment
+  @objc public init(userId: Swift.String, isAnonymous: Swift.Bool = false, rawId: Swift.String? = nil, cloudEnvironment: AzureCommunication.CommunicationCloudEnvironment = .Public)
+  @objc public init(userId: Swift.String, isAnonymous: Swift.Bool)
+  public static func == (lhs: AzureCommunication.MicrosoftTeamsUserIdentifier, rhs: AzureCommunication.MicrosoftTeamsUserIdentifier) -> Swift.Bool
+  @objc deinit
+  @objc override dynamic public init()
+}
+@objc @objcMembers public class CommunicationAccessToken : ObjectiveC.NSObject {
+  @objc final public let token: Swift.String
+  @objc final public let expiresOn: Foundation.Date
+  @objc public init(token: Swift.String, expiresOn: Foundation.Date)
+  @objc deinit
+  @objc override dynamic public init()
+}
+@objc @objcMembers public class CommunicationCloudEnvironment : ObjectiveC.NSObject {
+  @objc public static var Public: AzureCommunication.CommunicationCloudEnvironment
+  @objc public static var Dod: AzureCommunication.CommunicationCloudEnvironment
+  @objc public static var Gcch: AzureCommunication.CommunicationCloudEnvironment
+  @objc public init(environmentValue: Swift.String)
+  @objc public func getEnvironmentValue() -> Swift.String
+  public static func == (lhs: AzureCommunication.CommunicationCloudEnvironment, rhs: AzureCommunication.CommunicationCloudEnvironment) -> Swift.Bool
+  @objc deinit
+  @objc override dynamic public init()
+}
+public typealias TokenRefresherClosure = (@escaping AzureCommunication.TokenRefreshOnCompletion) -> Swift.Void
+@objc @objcMembers public class CommunicationTokenRefreshOptions : ObjectiveC.NSObject {
+  @objc public init(initialToken: Swift.String? = nil, refreshProactively: Swift.Bool = false, tokenRefresher: @escaping AzureCommunication.TokenRefresherClosure)
+  @objc deinit
+  @objc override dynamic public init()
+}
+public class CommunicationPolicyTokenCredential : AzureCore.TokenCredential {
+  public init(_ credential: AzureCommunication.CommunicationTokenCredential)
+  public func token(forScopes _: [Swift.String], completionHandler: @escaping AzureCore.TokenCompletionHandler)
+  public func validate() throws
+  @objc deinit
+}
+public typealias CommunicationTokenCompletionHandler = (AzureCommunication.CommunicationAccessToken?, Swift.Error?) -> Swift.Void
+public typealias TokenRefreshOnCompletion = (Swift.String?, Swift.Error?) -> Swift.Void
+@objc @objcMembers public class CommunicationTokenCredential : ObjectiveC.NSObject {
+  @objc public init(token: Swift.String) throws
+  @objc public init(with option: AzureCommunication.CommunicationTokenRefreshOptions) throws
```

Comment:
Looking at the guidelines, we always use `withOptions options:`. That's what I would recommend here. Also, is it possible to defer the `throwing` until the Credential object is used? With our Storage UX study, participants did not like when creating the credential could throw... and then creating the client with the credential could throw... so we consolidated toward making just the client creation fail, rather than the credential creation.

Reply:
I don't think we can defer the throwing here. Its a little complex but the `CommunicationTokenCredential` when trying to create a `StaticTokenCredential` uses the `JWTTokenParser`. If the token passed in does not conform to the correct jwt format we want to throw. The throw comes from the JWT validation essentially.

Reply:
I can update the method parameters, I will have to double check what the interface created for ObjC will be. It might look a little silly, so I will have and update it and make something specific for that case.

Reply:
+1 on what Josh said, Sever issues or connectivity, etc. will not cause exceptions here. Throwing only happens if the developers passes the wrong argument. In which case delaying the throw to the clients using it would burry the issue down and I think it would get very hard for developers to figure out what they did wrong.

Reply:
We talked offline. Considering that the exception is thrown during the initialization of the credential not the client, we can keep it as is